src/realm/group_writer.cpp

```diff
@@ -915,17 +940,27 @@ void GroupWriter::commit(ref_type new_top_ref)
     // stable storage before flipping the slot selector
     window->encryption_write_barrier(&file_header.m_top_ref[slot_selector],
                                      sizeof(file_header.m_top_ref[slot_selector]));
-    flush_all_mappings();
-    if (!disable_sync)
+    encryption_flush_all_private_caches();
+    if (!disable_sync) {
+        sync_all_mappings();
+#if REALM_PLATFORM_APPLE
```

Comment:
After testing this for quite a while, the barrier seems not needed. 
It seems that `msync()` issues a `barrier()` every time we sync a page, thus the order in which the pages are synced cannot be changed. At least this is the empirical evidence.
 Considering what core 11.8 introduced, with the barrier mechanism, this makes sense. Before core 11.8 we solely relied on `msync()` and 11.7 proved to be reliable and survive after a device restart.

Reply:
Since this is undocumented on Apple's part, we should probably keep the barrier, even if it seems like it is currently unneeded. They could choose to change the behaviour at any time.

We should obviously test if there is any performance difference, but if `msync` is already doing a barrier, doing a another on top should not be expected to cost more.

Reply:
I agree, I just wanted to point this out. I think `msync` is sufficient, however, since `F_BARRIERSYNC` is not blocking, I don't expect that to be a big performance penalty…